**Windows** [Build 4853](http://brontes.lab.eng.brq.redhat.com/viewLog.html?buildId=97857&buildTypeId=WF_PullRequest_Windows) outcome was **FAILURE** using a merge of afeb349f6defb2ea4eea8374d76f88ad84da1955
Summary: Tests failed: 2 (2 new), passed: 3312, ignored: 195 Build time: 00:58:43
### Failed tests
```
org.jboss.as.test.clustering.cluster.web.passivation.CoarseSessionPassivationTestCase(SYNC-tcp).test: java.lang.AssertionError: response doesn't contain header 'result', all response headers = {name=Connection, value=keep-alive}, {name=X-Powered-By, value=Undertow/1}, {name=Set-Cookie, value=JSESSIONID=oI9Uh8NqnQCa454R6nkckLnl0MiDzB0zMqtBIrOP.node-0; path=/passivation}, {name=Server, value=WildFly/10}, {name=vLFVEBncwG99A6YfJZxzIkq3ko3mScnHqRq1PiYs, value=PASSIVATION}, {name=Content-Length, value=0}, {name=jsessionid, value=oI9Uh8NqnQCa454R6nkckLnl0MiDzB0zMqtBIrOP}, {name=Date, value=Fri, 13 May 2016 07:47:53 GMT}, 
	at org.jboss.as.test.clustering.cluster.web.passivation.SessionPassivationTestCase.getRequiredHeaderValue(SessionPassivationTestCase.java:207)
	at org.jboss.as.test.clustering.cluster.web.passivation.SessionPassivationTestCase.test(SessionPassivationTestCase.java:153)
------- Stdout: -------
09:47:36,688 INFO  [org.jboss.as.test.clustering.NodeUtil] (main) Starting deployment=NONE, container=container-0
09:47:37,681 WARNING [org.jboss.as.arquillian.container.managed.ManagedDeployableContainer] (main) Bundles path is deprecated and no longer used.
09:47:37,682 INFO  [org.jboss.as.arquillian.container.managed.ManagedDeployableContainer] (main) Starting container with: [C:\Java\jdk1.8.0_66\bin\java, -D[Standalone], -Dorg.jboss.ejb.client.wildfly-testsuite-hack=true, -Xmx512m, -XX:MetaspaceSize=128m, -Djboss.dist=C:\BuildAgent\work\a31d203e70e89f90\testsuite\integration\clustering/../../../dist/target/wildfly-10.1.0.Final-SNAPSHOT, -Djava.net.preferIPv4Stack=false, -Djava.net.preferIPv6Addresses=true, -server, -Dts.timeout.factor=100, -Dnode0=::1, -Dnode1=::1, -Dmcast=ff13::1, -Dmcast.ttl=0, -Djbossas.ts.submodule.dir=C:\BuildAgent\work\a31d203e70e89f90\testsuite\integration\clustering, -Djbossas.ts.integ.dir=C:\BuildAgent\work\a31d203e70e89f90\testsuite\integration\clustering/.., -Djbossas.ts.dir=C:\BuildAgent\work\a31d203e70e89f90\testsuite\integration\clustering/../.., -Djbossas.project.dir=C:\BuildAgent\work\a31d203e70e89f90\testsuite\integration\clustering/../../.., -Djboss.inst=C:\BuildAgent\work\a31d203e70e89f90\testsuite\integration\clustering/target/wildfly-SYNC-tcp-0, -Djboss.node.name=node-0, -ea, -Djboss.home.dir=C:\BuildAgent\work\a31d203e70e89f90\testsuite\integration\clustering\target\wildfly-SYNC-tcp-0, -Dorg.jboss.boot.log.file=C:\BuildAgent\work\a31d203e70e89f90\testsuite\integration\clustering\target\wildfly-SYNC-tcp-0\standalone\log\server.log, -Dlogging.configuration=file:C:\BuildAgent\work\a31d203e70e89f90\testsuite\integration\clustering\target\wildfly-SYNC-tcp-0\standalone\configuration\logging.properties, -jar, C:\BuildAgent\work\a31d203e70e89f90\testsuite\integration\clustering\target\wildfly-SYNC-tcp-0\jboss-modules.jar, -mp, C:\BuildAgent\work\a31d203e70e89f90\testsuite\integration\clustering\target\wildfly-SYNC-tcp-0\modules;C:\BuildAgent\work\a31d203e70e89f90\dist\target\wildfly-10.1.0.Final-SNAPSHOT\modules;C:\BuildAgent\work\a31d203e70e89f90\testsuite\integration\clustering\target\modules, org.jboss.as.standalone, -Djboss.home.dir=C:\BuildAgent\work\a31d203e70e89f90\testsuite\integration\clustering\target\wildfly-SYNC-tcp-0, -Djboss.server.base.dir=C:\BuildAgent\work\a31d203e70e89f90\testsuite\integration\clustering\target\wildfly-SYNC-tcp-0\standalone, -Djboss.server.log.dir=C:\BuildAgent\work\a31d203e70e89f90\testsuite\integration\clustering\target\wildfly-SYNC-tcp-0\standalone\log, -Djboss.server.config.dir=C:\BuildAgent\work\a31d203e70e89f90\testsuite\integration\clustering\target\wildfly-SYNC-tcp-0\standalone\configuration, -Dts.wildfly.version=10.1.0.Final-SNAPSHOT, -c=standalone-full-ha.xml]
09:47:38,106 INFO  [org.jboss.modules] (main) JBoss Modules version 1.5.1.Final
09:47:38,292 INFO  [org.jboss.msc] (main) JBoss MSC version 1.2.6.Final
09:47:38,370 INFO  [org.jboss.as] (MSC service thread 1-6) WFLYSRV0049: WildFly Full 10.1.0.Final-SNAPSHOT (WildFly Core 2.2.0.CR2) starting
09:47:39,769 INFO  [org.jboss.as.server] (Controller Boot Thread) WFLYSRV0039: Creating http management service using socket-binding (management-http)
09:47:39,785 INFO  [org.xnio] (MSC service thread 1-4) XNIO version 3.3.6.Final
09:47:39,793 INFO  [org.xnio.nio] (MSC service thread 1-4) XNIO NIO Implementation Version 3.3.6.Final
09:47:39,856 INFO  [org.jboss.as.clustering.infinispan] (ServerService Thread Pool -- 44) WFLYCLINF0001: Activating Infinispan subsystem.
09:47:39,872 INFO  [org.jboss.as.jsf] (ServerService Thread Pool -- 52) WFLYJSF0007: Activated the following JSF Implementations: [main]
09:47:40,037 INFO  [org.jboss.as.naming] (ServerService Thread Pool -- 57) WFLYNAM0001: Activating Naming Subsystem
09:47:40,040 INFO  [org.jboss.as.security] (ServerService Thread Pool -- 64) WFLYSEC0002: Activating Security Subsystem
09:47:40,040 WARN  [org.jboss.as.txn] (ServerService Thread Pool -- 66) WFLYTX0013: Node identifier property is set to the default value. Please make sure it is unique.
09:47:40,054 INFO  [org.jboss.as.clustering.jgroups] (ServerService Thread Pool -- 49) WFLYCLJG0001: Activating JGroups subsystem.
09:47:40,035 INFO  [org.jboss.as.webservices] (ServerService Thread Pool -- 68) WFLYWS0002: Activating WebServices Extension
09:47:40,075 INFO  [org.jboss.as.security] (MSC service thread 1-7) WFLYSEC0001: Current PicketBox version=4.9.6.Final
09:47:40,100 INFO  [org.jboss.as.naming] (MSC service thread 1-8) WFLYNAM0003: Starting Naming Service
09:47:40,101 INFO  [org.jboss.as.mail.extension] (MSC service thread 1-8) WFLYMAIL0001: Bound mail session [java:jboss/mail/Default]
09:47:40,102 INFO  [org.wildfly.iiop.openjdk] (ServerService Thread Pool -- 45) WFLYIIOP0001: Activating IIOP Subsystem
09:47:40,102 INFO  [org.wildfly.extension.io] (ServerService Thread Pool -- 43) WFLYIO001: Worker 'default' has auto-configured to 8 core threads with 64 task threads based on your 4 available processors
09:47:40,117 INFO  [org.wildfly.extension.undertow] (MSC service thread 1-3) WFLYUT0003: Undertow 1.3.21.Final starting
09:47:40,128 INFO  [org.jboss.as.connector.subsystems.datasources] (ServerService Thread Pool -- 39) WFLYJCA0004: Deploying JDBC-compliant driver class org.h2.Driver (version 1.3)
09:47:40,142 INFO  [org.jboss.as.ejb3] (MSC service thread 1-3) WFLYEJB0481: Strict pool slsb-strict-max-pool is using a max instance size of 64 (per class), which is derived from thread worker pool sizing.
09:47:40,143 INFO  [org.jboss.as.ejb3] (MSC service thread 1-8) WFLYEJB0482: Strict pool mdb-strict-max-pool is using a max instance size of 16 (per class), which is derived from the number of CPUs on this host.
09:47:40,186 INFO  [org.jboss.as.connector] (MSC service thread 1-1) WFLYJCA0009: Starting JCA Subsystem (WildFly/IronJacamar 1.3.3.Final)
09:47:40,196 INFO  [org.jboss.as.connector.deployers.jdbc] (MSC service thread 1-8) WFLYJCA0018: Started Driver service with driver-name = h2
09:47:40,375 INFO  [org.jboss.remoting] (MSC service thread 1-4) JBoss Remoting version 4.0.18.Final
09:47:40,438 INFO  [org.wildfly.extension.undertow] (ServerService Thread Pool -- 67) WFLYUT0014: Creating file handler for path 'C:\BuildAgent\work\a31d203e70e89f90\testsuite\integration\clustering\target\wildfly-SYNC-tcp-0/welcome-content' with options [directory-listing: 'false', follow-symlink: 'false', case-sensitive: 'true', safe-symlink-paths: '[]']
09:47:40,495 INFO  [org.wildfly.extension.undertow] (MSC service thread 1-6) WFLYUT0012: Started server default-server.
09:47:40,496 INFO  [org.wildfly.extension.undertow] (MSC service thread 1-6) WFLYUT0018: Host default-host starting
09:47:40,570 INFO  [org.wildfly.extension.undertow] (MSC service thread 1-2) WFLYUT0006: Undertow HTTP listener default listening on [::1]:8080
09:47:40,575 INFO  [org.wildfly.extension.undertow] (MSC service thread 1-6) WFLYUT0006: Undertow AJP listener ajp listening on [::1]:8009
09:47:40,631 INFO  [org.jboss.modcluster] (ServerService Thread Pool -- 71) MODCLUSTER000001: Initializing mod_cluster version 1.3.2.Final
09:47:40,709 INFO  [org.jboss.modcluster] (ServerService Thread Pool -- 71) MODCLUSTER000032: Listening to proxy advertisements on /ff13:0:0:0:0:0:0:1:23364
09:47:40,995 INFO  [org.jboss.as.server.deployment.scanner] (MSC service thread 1-1) WFLYDS0013: Started FileSystemDeploymentService for directory C:\BuildAgent\work\a31d203e70e89f90\testsuite\integration\clustering\target\wildfly-SYNC-tcp-0\standalone\deployments
09:47:41,062 INFO  [org.jboss.as.connector.subsystems.datasources] (MSC service thread 1-5) WFLYJCA0001: Bound data source [java:jboss/datasources/ExampleDS]
09:47:41,065 INFO  [org.wildfly.iiop.openjdk] (MSC service thread 1-7) WFLYIIOP0009: CORBA ORB Service started
09:47:41,248 INFO  [org.jboss.ws.common.management] (MSC service thread 1-6) JBWS022052: Starting JBossWS 5.1.4.Final (Apache CXF 3.1.5) 
09:47:44,235 INFO  [org.wildfly.extension.messaging-activemq] (MSC service thread 1-4) WFLYMSGAMQ0001: AIO wasn't located on this platform, it will fall back to using pure Java NIO.
09:47:44,314 INFO  [org.apache.activemq.artemis.core.server] (ServerService Thread Pool -- 71) AMQ221000: live Message Broker is starting with configuration Broker Configuration (clustered=true,journalDirectory=C:\BuildAgent\work\a31d203e70e89f90\testsuite\integration\clustering\target\wildfly-SYNC-tcp-0\standalone\data\activemq\journal,bindingsDirectory=C:\BuildAgent\work\a31d203e70e89f90\testsuite\integration\clustering\target\wildfly-SYNC-tcp-0\standalone\data\activemq\bindings,largeMessagesDirectory=C:\BuildAgent\work\a31d203e70e89f90\testsuite\integration\clustering\target\wildfly-SYNC-tcp-0\standalone\data\activemq\largemessages,pagingDirectory=C:\BuildAgent\work\a31d203e70e89f90\testsuite\integration\clustering\target\wildfly-SYNC-tcp-0\standalone\data\activemq\paging)
09:47:44,375 INFO  [org.apache.activemq.artemis.core.server] (ServerService Thread Pool -- 71) AMQ221013: Using NIO Journal
09:47:44,547 INFO  [org.apache.activemq.artemis.core.server] (ServerService Thread Pool -- 71) AMQ221043: Protocol module found: [artemis-server]. Adding protocol support for: CORE
09:47:44,552 INFO  [org.apache.activemq.artemis.core.server] (ServerService Thread Pool -- 71) AMQ221043: Protocol module found: [artemis-amqp-protocol]. Adding protocol support for: AMQP
09:47:44,562 INFO  [org.apache.activemq.artemis.core.server] (ServerService Thread Pool -- 71) AMQ221043: Protocol module found: [artemis-hornetq-protocol]. Adding protocol support for: HORNETQ
09:47:44,564 INFO  [org.apache.activemq.artemis.core.server] (ServerService Thread Pool -- 71) AMQ221043: Protocol module found: [artemis-stomp-protocol]. Adding protocol support for: STOMP
09:47:44,615 INFO  [org.infinispan.remoting.transport.jgroups.JGroupsTransport] (MSC service thread 1-1) ISPN000078: Starting JGroups channel server


org.jboss.as.test.clustering.cluster.web.passivation.FineSessionPassivationTestCase(SYNC-tcp).test: java.lang.AssertionError: response doesn't contain header 'result', all response headers = {name=Connection, value=keep-alive}, {name=X-Powered-By, value=Undertow/1}, {name=Set-Cookie, value=JSESSIONID=Q-rsL0wT7eHznSUO0aOd2WfkZYzAYUlhuEbrl-_V.node-0; path=/passivation}, {name=Server, value=WildFly/10}, {name=Content-Length, value=0}, {name=jsessionid, value=Q-rsL0wT7eHznSUO0aOd2WfkZYzAYUlhuEbrl-_V}, {name=qgWhJFV9OtEzRYeB9e52fBigmyyMT2p4_Pc8SYzq, value=PASSIVATION}, {name=Date, value=Fri, 13 May 2016 07:47:54 GMT}, 
	at org.jboss.as.test.clustering.cluster.web.passivation.SessionPassivationTestCase.getRequiredHeaderValue(SessionPassivationTestCase.java:207)
	at org.jboss.as.test.clustering.cluster.web.passivation.SessionPassivationTestCase.test(SessionPassivationTestCase.java:153)
------- Stdout: -------
09:47:53,906 INFO  [org.jboss.as.test.clustering.NodeUtil] (main) Starting deployment=NONE, container=container-0
09:47:53,906 INFO  [org.jboss.as.test.clustering.NodeUtil] (main) Starting deployment=NONE, container=container-1
09:47:53,906 INFO  [org.jboss.as.test.clustering.NodeUtil] (main) Deploying deployment=deployment-0
09:47:53,923 INFO  [org.jboss.as.repository] (management-handler-thread - 3) WFLYDR0001: Content added at location C:\BuildAgent\work\a31d203e70e89f90\testsuite\integration\clustering\target\wildfly-SYNC-tcp-0\standalone\data\content\80\cb64f7e028ea0e7f824e0ab64120c53051a486\content
09:47:53,928 INFO  [org.jboss.as.server.deployment] (MSC service thread 1-1) WFLYSRV0027: Starting deployment of "passivation.war" (runtime-name: "passivation.war")
09:47:53,977 INFO  [org.infinispan.configuration.cache.EvictionConfigurationBuilder] (ServerService Thread Pool -- 27) ISPN000152: Passivation configured without an eviction policy being selected. Only manually evicted entities will be passivated.
09:47:53,980 INFO  [org.infinispan.configuration.cache.EvictionConfigurationBuilder] (ServerService Thread Pool -- 27) ISPN000152: Passivation configured without an eviction policy being selected. Only manually evicted entities will be passivated.
09:47:53,983 INFO  [org.infinispan.configuration.cache.EvictionConfigurationBuilder] (ServerService Thread Pool -- 27) ISPN000152: Passivation configured without an eviction policy being selected. Only manually evicted entities will be passivated.
09:47:53,984 INFO  [org.infinispan.configuration.cache.EvictionConfigurationBuilder] (ServerService Thread Pool -- 13) ISPN000152: Passivation configured without an eviction policy being selected. Only manually evicted entities will be passivated.
09:47:53,984 INFO  [org.infinispan.configuration.cache.EvictionConfigurationBuilder] (ServerService Thread Pool -- 27) ISPN000152: Passivation configured without an eviction policy being selected. Only manually evicted entities will be passivated.
09:47:53,985 INFO  [org.infinispan.configuration.cache.EvictionConfigurationBuilder] (ServerService Thread Pool -- 13) ISPN000152: Passivation configured without an eviction policy being selected. Only manually evicted entities will be passivated.
09:47:53,995 INFO  [org.jboss.as.clustering.infinispan] (ServerService Thread Pool -- 2) WFLYCLINF0002: Started routing cache from web container
09:47:54,002 INFO  [org.jboss.as.clustering.infinispan] (ServerService Thread Pool -- 13) WFLYCLINF0002: Started passivation.war cache from web container
09:47:54,017 INFO  [org.wildfly.extension.undertow] (ServerService Thread Pool -- 13) WFLYUT0021: Registered web context: /passivation
09:47:54,032 INFO  [org.jboss.as.server] (management-handler-thread - 3) WFLYSRV0010: Deployed "passivation.war" (runtime-name : "passivation.war")
09:47:54,038 INFO  [org.jboss.as.test.clustering.NodeUtil] (main) Deploying deployment=deployment-1
09:47:54,048 INFO  [org.jboss.as.repository] (management-handler-thread - 4) WFLYDR0001: Content added at location C:\BuildAgent\work\a31d203e70e89f90\testsuite\integration\clustering\target\wildfly-SYNC-tcp-1\standalone\data\content\f3\ebd513eacb5102a1838326b751d844f6bd15cc\content
09:47:54,051 INFO  [org.jboss.as.server.deployment] (MSC service thread 1-6) WFLYSRV0027: Starting deployment of "passivation.war" (runtime-name: "passivation.war")
09:47:54,109 INFO  [org.infinispan.configuration.cache.EvictionConfigurationBuilder] (ServerService Thread Pool -- 18) ISPN000152: Passivation configured without an eviction policy being selected. Only manually evicted entities will be passivated.
09:47:54,110 INFO  [org.infinispan.configuration.cache.EvictionConfigurationBuilder] (ServerService Thread Pool -- 18) ISPN000152: Passivation configured without an eviction policy being selected. Only manually evicted entities will be passivated.
09:47:54,111 INFO  [org.infinispan.configuration.cache.EvictionConfigurationBuilder] (ServerService Thread Pool -- 18) ISPN000152: Passivation configured without an eviction policy being selected. Only manually evicted entities will be passivated.
09:47:54,112 INFO  [org.infinispan.configuration.cache.EvictionConfigurationBuilder] (ServerService Thread Pool -- 18) ISPN000152: Passivation configured without an eviction policy being selected. Only manually evicted entities will be passivated.
09:47:54,121 INFO  [org.infinispan.configuration.cache.EvictionConfigurationBuilder] (ServerService Thread Pool -- 34) ISPN000152: Passivation configured without an eviction policy being selected. Only manually evicted entities will be passivated.
09:47:54,122 INFO  [org.infinispan.configuration.cache.EvictionConfigurationBuilder] (ServerService Thread Pool -- 34) ISPN000152: Passivation configured without an eviction policy being selected. Only manually evicted entities will be passivated.
09:47:54,124 INFO  [org.infinispan.CLUSTER] (remote-thread--p8-t2) ISPN000310: Starting cluster-wide rebalance for cache routing, topology CacheTopology{id=1, rebalanceId=1, currentCH=DefaultConsistentHash{ns=80, owners = (1)[node-0: 80+0]}, pendingCH=DefaultConsistentHash{ns=80, owners = (2)[node-0: 40+40, node-1: 40+40]}, unionCH=null, actualMembers=[node-0, node-1]}
09:47:54,136 INFO  [org.infinispan.CLUSTER] (remote-thread--p8-t2) ISPN000310: Starting cluster-wide rebalance for cache passivation.war, topology CacheTopology{id=1, rebalanceId=1, currentCH=DefaultConsistentHash{ns=80, owners = (1)[node-0: 80+0]}, pendingCH=DefaultConsistentHash{ns=80, owners = (2)[node-0: 40+40, node-1: 40+40]}, unionCH=null, actualMembers=[node-0, node-1]}
09:47:54,143 INFO  [org.infinispan.CLUSTER] (remote-thread--p8-t1) ISPN000336: Finished cluster-wide rebalance for cache routing, topology id = 1
09:47:54,147 INFO  [org.jboss.as.clustering.infinispan] (ServerService Thread Pool -- 18) WFLYCLINF0002: Started routing cache from web container
09:47:54,158 INFO  [org.infinispan.CLUSTER] (remote-thread--p8-t1) ISPN000336: Finished cluster-wide rebalance for cache passivation.war, topology id = 1
09:47:54,162 INFO  [org.jboss.as.clustering.infinispan] (ServerService Thread Pool -- 6) WFLYCLINF0002: Started passivation.war cache from web container
09:47:54,180 INFO  [org.wildfly.extension.undertow] (ServerService Thread Pool -- 6) WFLYUT0021: Registered web context: /passivation
09:47:54,198 INFO  [org.jboss.as.server] (management-handler-thread - 4) WFLYSRV0010: Deployed "passivation.war" (runtime-name : "passivation.war")
09:47:54,220 INFO  [stdout] (default task-6) http://[::1]:8080/passivation/listener?operation=set&name=a&value=1;jsessionid=qgWhJFV9OtEzRYeB9e52fBigmyyMT2p4_Pc8SYzq

09:47:54,227 INFO  [stdout] (default task-7) http://[::1]:8080/passivation/listener?operation=set&name=a&value=2;jsessionid=wuYytt1rlvHsqGB_GEDuGej5_8LWCkkgrpJ0AJ-y

09:47:54,238 INFO  [stdout] (default task-8) http://[::1]:8080/passivation/listener?operation=get&name=a;jsessionid=qgWhJFV9OtEzRYeB9e52fBigmyyMT2p4_Pc8SYzq

09:47:54,270 INFO  [stdout] (default task-9) http://[::1]:8080/passivation/listener?operation=get&name=a;jsessionid=Q-rsL0wT7eHznSUO0aOd2WfkZYzAYUlhuEbrl-_V

09:47:54,272 INFO  [org.jboss.as.test.clustering.NodeUtil] (main) Starting deployment=NONE, container=container-0
09:47:54,272 INFO  [org.jboss.as.test.clustering.NodeUtil] (main) Starting deployment=NONE, container=container-1
09:47:54,272 INFO  [org.jboss.as.test.clustering.NodeUtil] (main) Undeploying deployment=deployment-0
09:47:54,280 INFO  [org.wildfly.extension.undertow] (ServerService Thread Pool -- 13) WFLYUT0022: Unregistered web context: /passivation
09:47:54,284 INFO  [org.infinispan.eviction.impl.PassivationManagerImpl] (ServerService Thread Pool -- 27) ISPN000029: Passivating all entries to disk
09:47:54,285 INFO  [org.infinispan.eviction.impl.PassivationManagerImpl] (ServerService Thread Pool -- 27) ISPN000030: Passivated 2 entries in 0 milliseconds
09:47:54,288 INFO  [org.jboss.as.clustering.infinispan] (ServerService Thread Pool -- 27) WFLYCLINF0003: Stopped passivation.war cache from web container


```

--- a/pom.xml
+++ b/pom.xml
@@ -164,7 +164,7 @@
         <version.org.jboss.jandex>2.0.2.Final</version.org.jboss.jandex>
         <version.org.jboss.jboss-transaction-spi>7.3.0.Final</version.org.jboss.jboss-transaction-spi>
         <version.org.jboss.metadata>10.0.0.Final</version.org.jboss.metadata>
-        <version.org.jboss.narayana>5.2.16.Final</version.org.jboss.narayana>
+        <version.org.jboss.narayana>5.3.0.Final</version.org.jboss.narayana>
         <version.org.jboss.mod_cluster>1.3.2.Final</version.org.jboss.mod_cluster>
         <version.org.jboss.openjdk-orb>8.0.6.Final</version.org.jboss.openjdk-orb>
         <version.org.jboss.xnio.netty.netty-xnio-transport>0.1.1.Final</version.org.jboss.xnio.netty.netty-xnio-transport>

--- a/transactions/src/main/java/org/jboss/as/txn/service/JTAEnvironmentBeanService.java
+++ b/transactions/src/main/java/org/jboss/as/txn/service/JTAEnvironmentBeanService.java
@@ -24,6 +24,7 @@ package org.jboss.as.txn.service;
 
 import com.arjuna.ats.internal.jta.recovery.arjunacore.JTANodeNameXAResourceOrphanFilter;
 import com.arjuna.ats.internal.jta.recovery.arjunacore.JTATransactionLogXAResourceOrphanFilter;
+import com.arjuna.ats.internal.jta.recovery.arjunacore.JTAActionStatusServiceXAResourceOrphanFilter;
 import com.arjuna.ats.internal.jta.recovery.arjunacore.SubordinateJTAXAResourceOrphanFilter;
 import com.arjuna.ats.jta.common.JTAEnvironmentBean;
 import com.arjuna.ats.jta.common.jtaPropertyManager;
@@ -44,9 +45,11 @@ import java.util.Collections;
 public class JTAEnvironmentBeanService implements Service<JTAEnvironmentBean> {
 
     private final String nodeIdentifier;
+    private boolean useActionStatusServiceRecoveryFilter;
 
     public JTAEnvironmentBeanService(final String nodeIdentifier) {
         this.nodeIdentifier = nodeIdentifier;
+        this.useActionStatusServiceRecoveryFilter = Boolean.valueOf(System.getProperty("org.jboss.narayana.wildfly.useActionStatusServiceRecoveryFilter.deprecated", "false"));
     }
 
     @Override
@@ -56,7 +59,11 @@ public class JTAEnvironmentBeanService implements Service<JTAEnvironmentBean> {
         // recovery nodes
         jtaEnvironmentBean.setXaRecoveryNodes(Collections.singletonList(nodeIdentifier));
         // setup the XA orphan filters
-        jtaEnvironmentBean.setXaResourceOrphanFilterClassNames(Arrays.asList(JTATransactionLogXAResourceOrphanFilter.class.getName(), JTANodeNameXAResourceOrphanFilter.class.getName(), SubordinateJTAXAResourceOrphanFilter.class.getName()));
+        if (useActionStatusServiceRecoveryFilter) {
+            jtaEnvironmentBean.setXaResourceOrphanFilterClassNames(Arrays.asList(JTATransactionLogXAResourceOrphanFilter.class.getName(), JTANodeNameXAResourceOrphanFilter.class.getName(), SubordinateJTAXAResourceOrphanFilter.class.getName(), JTAActionStatusServiceXAResourceOrphanFilter.class.getName()));
+        } else {
+            jtaEnvironmentBean.setXaResourceOrphanFilterClassNames(Arrays.asList(JTATransactionLogXAResourceOrphanFilter.class.getName(), JTANodeNameXAResourceOrphanFilter.class.getName(), SubordinateJTAXAResourceOrphanFilter.class.getName()));
+        }
         jtaEnvironmentBean.setXAResourceRecordWrappingPlugin(new com.arjuna.ats.internal.jbossatx.jta.XAResourceRecordWrappingPluginImpl());
         jtaEnvironmentBean.setTransactionManagerJNDIContext("java:jboss/TransactionManager");
         jtaEnvironmentBean.setTransactionSynchronizationRegistryJNDIContext("java:jboss/TransactionSynchronizationRegistry");

